File: examples/python/capture_snapshot.py
"""Example: Capture a snapshot from an OpenClaw Gateway instance."""
import asyncio
from minions_openclaw import InstanceManager, SnapshotManager, GatewayClient


async def main():
    instance_manager = InstanceManager()
    snapshot_manager = SnapshotManager()

    instances = instance_manager.list()
    if not instances:
        print('No instances found. Register one first.')
        return

    instance = instances[0]
    print(f'Capturing snapshot for: {instance.title}')

    client = GatewayClient(
        instance.fields['url'],
        token=instance.fields.get('token'),
        device_private_key=instance.fields.get('devicePrivateKey'),
    )

    await client.open_connection()
    presence = await client.fetch_presence()
    await client.close()

    snapshot = snapshot_manager.capture_snapshot(instance.id, presence)
    print(f'Snapshot captured: {snapshot.id}')
    print(f"Agents: {presence['agents']}")


if __name__ == '__main__':
    asyncio.run(main())

File: examples/typescript/capture-snapshot.ts
/**
 * Example: Capture a snapshot from an OpenClaw Gateway instance
 */
import { InstanceManager, SnapshotManager, GatewayClient } from '@minions-openclaw/sdk';

async function main() {
  const instanceManager = new InstanceManager();
  const snapshotManager = new SnapshotManager();

  const instances = await instanceManager.list();
  if (instances.length === 0) {
    console.log('No instances found. Register one first.');
    return;
  }

  const instance = instances[0];
  console.log('Capturing snapshot for:', instance.title);

  const client = new GatewayClient(
    instance.fields['url'] as string,
    instance.fields['token'] as string | undefined,
    instance.fields['devicePrivateKey'] as string | undefined
  );

  await client.openConnection();
  const presence = await client.fetchPresence();
  await client.close();

  const snapshot = await snapshotManager.captureSnapshot(instance.id, presence);
  console.log('Snapshot captured:', snapshot.id);
  console.log('Agents:', presence.agents.length);
  console.log('Channels:', presence.channels.length);
  console.log('Models:', presence.models.length);
}

main().catch(console.error);
